packages/sanity/src/core/store/_legacy/document/document-pair/operationArgs.ts

```diff
@@ -20,27 +20,40 @@ export const operationArgs = memoize(
       historyStore: HistoryStore
       schema: Schema
       serverActionsEnabled: Observable<boolean>
+      bundlePerspective?: string
     },
     idPair: IdPair,
     typeName: string,
   ): Observable<OperationArgs> => {
-    // TODO: Should be dynamic
-    const draftIndex = 0
     return snapshotPair(ctx.client, idPair, typeName, ctx.serverActionsEnabled).pipe(
       switchMap((versions) =>
         combineLatest([
-          versions.drafts[draftIndex].snapshots$,
+          versions.draft.snapshots$,
           versions.published.snapshots$,
           ctx.serverActionsEnabled,
+          ...(typeof versions.version === 'undefined' ? [] : [versions.version.snapshots$]),
```

Comment:
Using an empty array here will effectively create an observable that never emits a value, so I think what you want here is rather `of()` (which will emit `undefined`)

Reply:
Ahhh, I didn't consider the spread – you are of course right, nvm :)